### PR TITLE
feature(wyx): add three KL-divergence variants

### PR DESCRIPTION
**Pull request by xiaoyi1734**
## Description

- Add KL-divergence and its variants to the standard PPO training process.

## Check List

- [ ] merge the latest version source branch/repo, and resolve all the conflicts
- [ ] pass style check
- [ ] pass all the tests

### CODE REVIEW

File: ding/rl_utils/ppo.py [hunk 1]
```diff
@@ -57,6 +87,7 @@ def ppo_error(
         - use_value_clip (:obj:`bool`): whether to use clip in value loss with the same ratio as policy
         - dual_clip (:obj:`float`): a parameter c mentioned in arXiv:1912.09729 Equ. 5, shoule be in [1, inf),\
         defaults to 5.0, if you don't want to use it, set this parameter to None
+        - kl_type (:obj:`str`): which kl loss to use, default set to 'approx'.
```
Comment (PaParaZz1) [inline ding/rl_utils/ppo.py:90]: why use `approx` here

File: ding/rl_utils/ppo.py [hunk 2]
```diff
@@ -180,7 +213,16 @@ def ppo_policy_error(
...
+
+    if logit_pretrained is not None:
+        dist_pretrained = torch.distributions.categorical.Categorical(logits=logit_pretrained)
+        logp_pretrained = dist_pretrained.log_prob(action)
+        logr = logp_new - logp_pretrained
+        kl_div = calculate_kl_div(logr, kl_type)
+    else:
+        kl_div = 0
+
+    return ppo_policy_loss(policy_loss, entropy_loss), ppo_info(approx_kl, clipfrac, kl_div)
```
Comment (PaParaZz1) [inline ding/rl_utils/ppo.py:225]: `kl_div` should be one of the output item of `ppo_policy_loss`, `ppo_info` is just the monitor variable

File: dizoo/atari/config/serial/spaceinvaders/spaceinvaders_onppo_config.py [hunk 3]
```diff
@@ -44,6 +44,9 @@
             ignore_done=False,
             grad_clip_type='clip_norm',
             grad_clip_value=0.5,
+            kl_beta=0.05,
+            kl_type='k1',
+            pretrained_model_path='The path of your pretrained model',
```
Comment (PaParaZz1) [inline dizoo/atari/config/serial/spaceinvaders/spaceinvaders_onppo_config.py:49]: <path-to-your-pretrained-model>

File: ding/policy/ppo.py [hunk 4]
```diff
@@ -76,6 +76,15 @@ class PPOPolicy(Policy):
             grad_clip_value=0.5,
             # (bool) Whether ignore done (usually for max step termination env).
             ignore_done=False,
+            # (str) The type of KL divergence loss, ['k1', 'k2', 'k3'].
```
Comment (PaParaZz1) [inline ding/policy/ppo.py:79]: The type of KL divergence loss between current policy and pretrained policy

File: ding/policy/ppo.py [hunk 5]
```diff
@@ -593,6 +626,7 @@ def _monitor_vars_learn(self) -> List[str]:
             'clipfrac',
             'value_max',
             'value_mean',
+            'kl_div',
```
Comment (PaParaZz1) [inline ding/policy/ppo.py:629]: if self._pretrained_model is not None, then return `kl_div` for monitoring.

**Review by PaParaZz1 — APPROVED** [review]
